lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -74,24 +97,45 @@ export default function LMSFilePicker({
         authToken,
         path: listFilesApi.path,
       }));
-      setDialogState({ ...INITIAL_DIALOG_STATE, state: 'fetched', files });
+      const continueAction =
+        files.length === 0 ? 'none' : INITIAL_DIALOG_STATE.continueAction;
+      setDialogState({
+        ...INITIAL_DIALOG_STATE,
+        state: 'fetched',
+        files,
+        continueAction,
+      });
     } catch (e) {
       if (e instanceof ApiError && !e.errorMessage) {
+        const continueAction = authorizationAttempted
+          ? 'authorize_retry'
+          : 'authorize';
+
         // If the server returned an error, but provided no details, assume
         // an authorization failure.
-        setDialogState({ ...INITIAL_DIALOG_STATE, state: 'authorizing' });
+        setDialogState({
+          ...INITIAL_DIALOG_STATE,
+          state: 'authorizing',
+          title: 'Allow file access',
+          continueAction,
+        });
+        setAuthorizationAttempted(true);
       } else {
         // Otherwise, display the error to the user.
-        setDialogState({ ...INITIAL_DIALOG_STATE, state: 'error', error: e });
+        setDialogState({
+          ...INITIAL_DIALOG_STATE,
+          state: 'error',
+          title: 'Error accessing files',
+          error: e,
+          continueAction: 'retry',
+        });
       }
     }
-  }, [authToken, listFilesApi]);
+  }, [authToken, listFilesApi, authorizationAttempted]);
 
   // Execute the authorization flow in a popup window and then attempt to
   // fetch files.
   const authorizeAndFetchFiles = useCallback(async () => {
-    setDialogState({ ...INITIAL_DIALOG_STATE, state: 'authorizing' });
```

Comment:
There are still changes to the `authorizeAndFetchFiles` logic here. Are these still applicable or do they need to be reverted?

Reply:
No, they don't need to be reverted.

The 'authorizing' state is set within the `fetchFiles` function, when an error is raised from the API.